docs/proposals/20251001-rag-auto-indexer.md

```diff
@@ -0,0 +1,592 @@
+
+	// Static defines configuration for static data sources
+	// +optional
+	Static *StaticDataSourceSpec `json:"static,omitempty"`
+}
+
+// DataSourceType defines the supported data source types
+// +kubebuilder:validation:Enum=Git;Static
+type DataSourceType string
+
+const (
+	DataSourceTypeGitHub DataSourceType = "Git"
+	DataSourceTypeStatic DataSourceType = "Static"
+)
+
+// GitHubDataSourceSpec defines GitHub repository configuration
+type GitDataSourceSpec struct {
+	// Repository to index. If the repository is not public and a token is needed for access,
+	// the access token can be stored in a secret and loaded with the SecretRef in the credential spec
+	// +kubebuilder:validation:Required
+	Repository string `json:"repository"`
+
+	// Branch to checkout (default: main)
+	// +kubebuilder:validation:Required
+	Branch string `json:"branch"`
+
+	// Commit SHA to checkout. If included, only this commit will be put into the index
+	// +optional
+	Commit *string `json:"commit,omitempty"`
+
+	// Specific paths to index within the repository.
+	// Can be directories, specific files, or specific extension types: /src, main.py, *.go
+	// +optional
+	Paths []string `json:"paths,omitempty"`
+
+	// Paths to exclude from indexing. ExcludePaths takes priority over Paths.
+	// Can be directories, specific files, or specific extension types: /src, main.py, *.go
+	// +optional
+	ExcludePaths []string `json:"excludePaths,omitempty"`
+}
+
+// APIDataSourceSpec defines REST API configuration
+type StaticDataSourceSpec struct {
+	// URLs that should point to individual text encoding (UTF-8, UTF-8-SIG, Latin1, etc) or pdf files.
+	// If an access token is needed for the URL's, the access token can be stored in a secret
+	// and loaded with the SecretRef in the credential spec
+	// +kubebuilder:validation:Required
+	URLs []string `json:"urls"`
+}
+
+// CredentialsSpec defines authentication credentials
+type CredentialsSpec struct {
+	// Type specifies the credential type
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=SecretRef
+	Type CredentialType `json:"type"`
+
+	// Secret reference containing credentials
+	// +optional
+	SecretRef *SecretKeyRef `json:"secretRef,omitempty"`
+}
+
+// CredentialType defines the supported credential types
+// +kubebuilder:validation:Enum=SecretRef
+type CredentialType string
+
+const (
+	CredentialTypeSecretRef CredentialType = "SecretRef"
+)
+
+// SecretKeyRef references a key in a Secret
+type SecretKeyRef struct {
+	// Secret name
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Key within the secret
+	// +kubebuilder:validation:Required
+	Key string `json:"key"`
+}
+
+// AutoIndexerStatus defines the observed state of AutoIndexer
+type AutoIndexerStatus struct {
+	// LastIndexedTimespamp is the timestamp of the end of the last successful indexing
+	// +optional
+	LastIndexedTimespamp *metav1.Time `json:"lastIndexedTimestamp,omitempty"`
```

Comment:
LastIndexedTimestamp -> LastIndexingTimestamp.

Reply:
renamed the field